### DISCUSSION TITLE
Per-app keybinds possible?

### DISCUSSION THREAD

**Discussion by hotsno** [hyprwm/Hyprland#1786] [Q&A]
I want to rebind the keybinds in some apps that don't have any built in options to change their keybinds. 

I don't think this is can be done with any built in Hyprland features, but I'm wondering if something like this is even possible in Wayland? I've seen some discussion that this is made difficult/impossible in Wayland by design. But if it is possible, would Hyprland be the appropriate place to implement something like this?

This would require 2 things that aren't currently possible in Hyprland though, I guess: the ability the set keybinds on a per-app basis, and the ability to map a keybind to another keybind.

This would give the user full control over their keybinds in every app.

**Comment by vaxerski** [comment]
plugin I guess

**Comment by ErfanRasti** [comment]
I think there is a possible way now. You can define a submap keybindings and trigger it using a tiny daemon as follows: create a script that follows whether you are currently focused on a specific app or not, and if true, it will trigger the submap, and you'll have all your keybindings. You can add this script to your `exec-once` or enable it using `systemd` services if you're using `systemd` or `uwsm`, probably.
